sky/execution.py

```diff
@@ -451,15 +456,47 @@ def launch(
         controller_utils.check_cluster_name_not_controller(
             cluster_name, operation_str='sky.launch')
 
+    handle = None
+    stages = None
+    # Check if cluster exists and we are doing fast provisioning
+    if fast and cluster_name is not None:
+        maybe_handle = global_user_state.get_handle_from_cluster_name(
+            cluster_name)
+        if maybe_handle is not None:
+            try:
+                # This will throw if the cluster is not available
+                backend_utils.check_cluster_available(
+                    cluster_name,
+                    operation='executing tasks',
+                    check_cloud_vm_ray_backend=False,
+                    dryrun=dryrun)
+                # If the cluster is available, restrict stages
+                handle = maybe_handle
+                stages = [
+                    # Stage.CLONE_DISK,
+                    # Stage.PROVISION,
+                    # Stage.OPTIMIZE,
+                    Stage.SYNC_WORKDIR,
+                    # Stage.SYNC_FILE_MOUNTS,
+                    # Stage.SETUP,
+                    # Stage.PRE_EXEC,
+                    Stage.EXEC,
+                    # Stage.DOWN
+                ]
```

Comment:
Nice to have the steps commented, but it may fall out of sync if we add/remove steps... best to just list what's included:
```suggestion
                stages = [
                    Stage.SYNC_WORKDIR,
                    Stage.EXEC,
                ]
```

Reply:
Will just revamp this to get the full list of stages and remove the ones we don't want, since that more accurately captures the goal.

Reply:
I think it's clearer to be explicit about what's actually run, rather than what's not run. This is also consistent with how we [specify stages in exec](https://github.com/cg505/skypilot/blob/d602225c897f10ca67a2d5b5db21982c0dc8c1ec/sky/execution.py#L561C25-L565C11). Let's have

```
                stages = [
                    Stage.SYNC_WORKDIR,
                    Stage.SYNC_FILE_MOUNTS
                    Stage.EXEC,
                ]
```

Reply:
I think it's clearer to be explicit about what's actually run, rather than what's not run. This is also consistent with how we [specify stages in exec](https://github.com/cg505/skypilot/blob/d602225c897f10ca67a2d5b5db21982c0dc8c1ec/sky/execution.py#L561C25-L565C11). Let's have

```
                stages = [
                    Stage.SYNC_WORKDIR,
                    Stage.SYNC_FILE_MOUNTS
                    Stage.EXEC,
                ]
```